### PR TITLE
Dota 2 Insight Hub Project Summary

### PR DESCRIPTION
**Pull request by lily4178993**
# Dota 2 Insight Hub Project Summary

## Overview

This pull request represents the culmination of work on the Dota 2 Insight Hub project. The project includes features related to heroes, items, pro matches, and pro players, and is designed to be a valuable resource for the Dota 2 community.

### Home Page

- The home page provides a clear and responsive user interface.
- Users can explore four main categories: heroes, items, pro matches, and pro players.
- The filter feature allows sorting data by names or counts.

### Details Page

- Clicking on an item redirects users to a detailed page for heroes, items, matches, or players.
- Users can access extensive data, including hero stats, item attributes, match details, and player profiles.
- The design is user-friendly, ensuring a seamless navigation experience.

### Performant Search

- Real-time data updates to keep users informed with the latest Dota 2 information.

### Testing

- Thorough testing has been conducted to ensure the reliability of the project.
- Tests have been implemented for various components, including Filters, References, DetailsItem, and Home, using snapshot testing.

## Implementation

- Redux has been used for state management to create a streamlined and efficient user experience.
- The project connects to external APIs, fetching data from Dota 2 to provide up-to-date information.
- Responsive design ensures that the project functions seamlessly on various devices and screen sizes.
- Extensive data parsing has been implemented to provide rich content and insights.

## Testing

- Comprehensive tests have been developed to validate the functionality of key components.
- Snapshot testing ensures the stability and consistency of components across different states.

## Future Enhancements

- The project has been built with scalability in mind, leaving room for future features and improvements.

## Pull Request Details

- Commits in this pull request represent the finalization of the Dota 2 Insight Hub project.
- Merged feature branches have been thoroughly reviewed and tested to ensure quality.

## Reviewer Notes:

Please review the entire codebase for adherence to best practices, error handling, and consistent behavior. Confirm that the application functions correctly and reliably with the API.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,347 @@
...
-In the project directory, you can run:
+<!-- TABLE OF CONTENTS -->
 
-### `npm start`
+# 📗 Table of Contents
 
-Runs the app in the development mode.\
-Open [http://localhost:3000](http://localhost:3000) to view it in your browser.
+- [📖 About the Project](#about-project)
+  - [📹 Video Presentation](#video-presentation)
```
Comment (shella12) [inline README.md:14]: - [ ]  There is a problem with your presentation link and its not opening. See the screenshot
Screenshot 2023-10-14 at 6 29 28 PM

File: src/__tests__/__snapshots__/ItemPresentation.test.jsx.snap [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+         Sample Item
+      </p>
+      <p>
+        <strong>
+          Type:
+        </strong>
+         Sample Quality
+      </p>
+      <p>
+        <strong>
```
Comment (shella12) [inline src/__tests__/__snapshots__/ItemPresentation.test.jsx.snap:27]: - [ ] [OPTIONAL] Great job on your app your UI is great and the app is cool however there are some cards that upon clicking leads to an error and the error is caused by empty array iteration. I would suggest handling  this error by putting a checks before iterating any array to see if it contains any item before iterating it. See screenshot of error:

Screenshot 2023-10-14 at 6 31 50 PM

**Review by shella12 — CHANGES_REQUESTED** [review]
Hi @lily4178993, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- Deployed the app ✅ 
- Added `Acknowledgments` in README ✅   
- Followed UI design guidelines ✅ 
- Well organised project structure ✅ 
- All tests are passing ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__tests__/__snapshots__/ItemPresentation.test.jsx.snap (hunk 2 above)
Comment (lily4178993) [reply src/__tests__/__snapshots__/ItemPresentation.test.jsx.snap:27]: I will implement this in the near future.

**Comment by lily4178993** [comment]
Thank you @shella12 for your review and feedbacks

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @lily4178993👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Nice UI design. ✔️
- Highly organized code. ✔️
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by lily4178993** [comment]
Thank you @thecodechaser for your review and feedback.

**Pull request merged**